component/otelcol/exporter/prometheus/internal/convert/convert_test.go

```diff
@@ -197,6 +199,100 @@ func TestConverter(t *testing.T) {
+							"exponential_histogram": {
+								"aggregation_temporality": 2,
+								"data_points": [{
+									"start_time_unix_nano": 1000000000,
+									"time_unix_nano": 1000000000,
+									"scale": 0,
+									"count": 11,
+									"sum": 158.63,
+									"positive": {
+										"offset": -1,
+										"bucket_counts": [2, 1, 3, 2, 0, 0, 3]
+									},
+									"exemplars":[
+										{
+											"time_unix_nano": 1000000001,
+											"as_double": 3.0,
+											"span_id": "aaaaaaaaaaaaaaaa",
+											"trace_id": "aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa"
+										},
+										{
+											"time_unix_nano": 1000000003,
+											"as_double": 1.0,
+											"span_id": "cccccccccccccccc",
+											"trace_id": "cccccccccccccccccccccccccccccccc"
+										},
+										{
+											"time_unix_nano": 1000000002,
+											"as_double": 1.5,
+											"span_id": "bbbbbbbbbbbbbbbb",
+											"trace_id": "bbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbb"
+										}
+									]
+								}]
+							}
+						}]
+					}]
+				}]
+			}`,
+			// Native histograms don't have text format yet so we compare the string representation.
+			// The tests only allow one exemplar/series because it uses a map[series]exemplar as storage. Therefore only the exemplar "bbbbbbbbbbbbbbbb" is stored.
+			expect:            `sample_count:11 sample_sum:158.63 bucket:{exemplar:{label:{name:"span_id" value:"bbbbbbbbbbbbbbbb"} label:{name:"trace_id" value:"bbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbb"} value:1.5}} schema:0 zero_threshold:1e-128 zero_count:0 positive_span:{offset:0 length:7} positive_delta:2 positive_delta:-1 positive_delta:2 positive_delta:-1 positive_delta:-2 positive_delta:0 positive_delta:3`,
+			enableOpenMetrics: true,
+		},
+		{
+			name: "Exponential Histogram 2",
+			input: `{
+				"resource_metrics": [{
+					"scope_metrics": [{
+						"metrics": [{
+							"name": "test_exponential_histogram_2",
+							"exponential_histogram": {
+								"aggregation_temporality": 2,
+								"data_points": [{
+									"start_time_unix_nano": 1000000000,
+									"time_unix_nano": 1000000000,
+									"scale": 2,
+									"count": 19,
+									"sum": 200,
+									"zero_count" : 5,
+									"zero_threshold": 0.1,
+									"positive": {
+										"offset": 3,
+										"bucket_counts": [0, 0, 0, 0, 2, 1, 1, 0, 3, 0, 0]
+									},
+									"negative": {
+										"offset": 0,
+										"bucket_counts": [0, 4, 0, 2, 3, 0, 0, 3]
+									},
+									"exemplars":[
+										{
+											"time_unix_nano": 1000000001,
+											"as_double": 3.0,
+											"span_id": "aaaaaaaaaaaaaaaa",
+											"trace_id": "aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa"
+										}
+									]
+								}]
+							}
+						}]
+					}]
+				}]
+			}`,
+			// Native histograms don't have text format yet so we compare the string representation.
+			// zero_threshold is set to 1e-128 because dp.ZeroThreshold() is not yet available.
```

Comment:
Good question, we have an integration test that scrapes Prometheus native histogram and stores them in Mimir.
Looking at the '/metrics' endpoint of the script that generates the native histogram I can see that the native histogram is displayed as if it would be a conventional histogram (which is wrong):
```
# TYPE golang_native_histogram histogram
golang_native_histogram_bucket{address="0.0.0.0",port="9001",le="+Inf"} 21
golang_native_histogram_sum{address="0.0.0.0",port="9001"} 10503.816145252378
golang_native_histogram_count{address="0.0.0.0",port="9001"} 21
```
The reason is that the text format does not support the native histogram and treats it as a conventional histogram: https://github.com/prometheus/common/blob/main/expfmt/text_create.go

We can see that this is just the text format and that the histogram is properly scraped and stored as a native histogram in Mimir when we query it:
`{"status":"success","data":{"resultType":"vector","result":[{"metric":{"__name__":"golang_native_histogram","address":"0.0.0.0","instance":"localhost:9001","job":"prometheus.scrape.scrape_prom_metrics","port":"9001","test_name":"scrape_prom_metrics"},"histogram":[1699545965,{"count":"65","sum":"32193.804117909098","buckets":[[0,"13.45434264405943","14.672064691274738","1"] ...`

The native histogram description is defined here: https://prometheus.io/docs/prometheus/latest/querying/api/#native-histograms

Reply:
Thank you, but I'm still a bit confused. If Prometheus can scrape native histograms using the conventional histogram approach, and infer that this is a native histogram, then would Prometheus ever make a special text representation for a native histogram? Is there an open issue for it? If not, then maybe we should have a different test function which is specific to native histograms and compares all the native histogram information?

BTW, do you think it'd be better to stream out the native histograms's proto contents as a Json? And have the `expect` variable contain Json? This has a few advantages:
* It is more reliable than the `String()` function. `String()` is probably mostly meant for debugging?
* Comparing Json in tests gives more readable diffs. E.g.: `require.JSONEq(t, expectedNativeHistJson, actualNativeHistJson)`

Reply:
I could not find anything saying that the text format would be available any time soon. Comparing the JSON formats is a good idea, it makes the test much easier to read. I made the changes